README.md

```diff
@@ -6,27 +6,61 @@
+we'll be working on a front-end for an animal shelter. Sadly, there still are way too many cute pets without any owners.
+Let's help them out by creating a UI in React!
 
-This lab provides good opportunities for introducing `shouldComponentUpdate()`
-and `componentDidMount()` (for initialization) for example.
+On a high level, you will be working on several components that form the UI of the animal shelter adoption application.
+Users can filter for pets by type, and can adopt a pet of their choosing. Once a pet is adopted, they cannot un-adopt
+it. No backsies!
 
-I leave the narrative domain up to you, but the main thing is having students
-write actions that use `fetch()` to call an API, and that then use `setState()`
-to render the response (and pass parts of it down as props as necessary).
+There are several components that need your attention. All of these components can be found in the `components/` folder.
+Starting with the smallest one and working our way up:
 
-This lab is meant to be pretty challenging.
+### `Pet`
+1. Should have a `pet` prop. Use the attributes in this data to render the pet card correctly. It
+should show the pet's `name`, `type`, `age` and `weight`. Based on the pet's `type`, the component also needs to contain
+either a male (`♂`) or female (`♀`) symbol.
+2. Should have an `isAdopted` prop. Using this prop, render the correct button in the pet's card: if
+the pet is adopted, show the disabled button. Otherwise, show the primary button to adopt the pet.
+2. Should have an `onAdoptPet` callback prop. This callback prop gets called with the pet's `id` when the user clicks
+the adopt pet button — _not_ when they click the disabled button!
 
-## Resources
+### `PetBrowser`
+1. Should have a `pets` prop. This is an array of pets that the component uses to render `<Pet />` components.
+1. Should have an `onAdoptPet` prop. This callback prop gets passed to its `<Pet />` children components.
+1. Should have an `adoptedPets` prop. Use this prop to figure out if a pet is adopted or not, and pass that result to
+the `<Pet />` components in the form of an `isAdopted` prop.
+
+### `Filters`
+1. Should have a `filters` prop that contains all of the filters in the `App` state (see below). Hook up this value
+to the select component.
+2. Should have an `onChangeType` callback prop. This callback prop gets called whenever the value of the `<select>`
+element changes. It calls the `onChangeType` callback prop with one argument: the value of the select.
+3. Should have an `onFindPetsClick` callback prop. This callback prop gets called when the users clicks the 'Find pets'
+button.
+
+### `App`
+1. The app's initial state is already defined. Pass the right state properties to the `<Filters />` and `<PetBrowser />`
+components.
+2. When the `<Filters />` component calls the `onChangeType` prop you pass into it, the state needs to be updated
```

Comment:
Just to avoid ambiguity: "When the `<Filters />` component calls the `onChangeType` prop you pass into it, the state needs to be..." -> "When the `<Filters />` component calls the `onChangeType` prop you pass into it, the App's state needs to be..."